src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -947,8 +948,8 @@ ByteBuffer overlapDetection(ByteBuffer src, ByteBuffer dst) {
                 // side, we are not in overlap.
                 // NOTE: inPlaceArray does not apply here as direct buffers run
                 // through a byte[] to get to the combined intrinsic
-                if (((DirectBuffer) src).address() - srcaddr + src.position() >=
-                    ((DirectBuffer) dst).address() - dstaddr + dst.position()) {
+                if (NIO_ACCESS.getBufferAddress(src) - srcaddr + src.position() >=
+                        NIO_ACCESS.getBufferAddress(dst) - dstaddr + dst.position()) {
```

Comment:
```suggestion
                if (NIO_ACCESS.getBufferAddress(src) - srcaddr + src.position() >=
                    NIO_ACCESS.getBufferAddress(dst) - dstaddr + dst.position()) {
```
Here you can use the same alignment style as before